cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,344 @@ def preferred_ntp_clients(self):
+            # that can go right through to the command.
+            kwargs["groups"] = ",".join(groups)
+
+            primary_group = kwargs.get("primary_group")
+            if primary_group:
+                groups.append(primary_group)
+
+        if create_groups and groups:
+            for group in groups:
+                if not util.is_group(group):
+                    self.create_group(group)
+                    LOG.debug("created group '%s' for user '%s'", group, name)
+        if "uid" in kwargs:
+            kwargs["uid"] = str(kwargs["uid"])
+
+        unsupported_busybox_values: Dict[str, Any] = {
+            "groups": None,
+            "expiredate": None,
+            "inactive": None,
+            "passwd": None,
+        }
+
+        # Check the values and create the command
+        for key, val in sorted(kwargs.items()):
+            if key in adduser_opts and val and isinstance(val, str):
+                adduser_cmd.extend([adduser_opts[key], val])
+            elif (
+                key in unsupported_busybox_values
+                and val
+                and isinstance(val, str)
+            ):
+                # Busybox's 'adduser' does not support specifying these
+                # options so store them for use via alternative means.
+                if key == "groups":
+                    unsupported_busybox_values[key] = val.split(",")
+                else:
+                    unsupported_busybox_values[key] = val
+            elif key in adduser_flags and val:
+                adduser_cmd.append(adduser_flags[key])
+
+        # Don't create the home directory if directed so
+        # or if the user is a system user
+        if kwargs.get("no_create_home") or kwargs.get("system"):
+            adduser_cmd.append("-H")
+
+        # Busybox's 'adduser' puts username at end of command
+        adduser_cmd.append(name)
+
+        # Run the command
+        LOG.debug("Adding user %s", name)
+        try:
+            subp.subp(adduser_cmd)
+        except Exception as e:
+            LOG.warning("Failed to create user %s", name)
+            raise e
+
+        # Process remaining options that Busybox's 'adduser' does not support
+
+        addn_groups = unsupported_busybox_values["groups"]
+        if addn_groups is not None:
+            # Separately add user to each additional group as Busybox's
+            # 'adduser' does not support specifying additional groups.
+            for addn_group in addn_groups:  # pylint: disable=E1133
+                LOG.debug("Adding user to group %s", addn_group)
+                try:
+                    subp.subp(["addgroup", name, addn_group])
+                except subp.ProcessExecutionError as e:
+                    util.logexc(
+                        LOG,
+                        "Failed to add user %s to group %s",
+                        name,
+                        addn_group,
+                    )
+                    raise e
```

Comment:
Here we can drop the if addn_groups and  left-shift this conditional scope if we initialize unsupported_busybox_groups["goups"] to an empty list instead of None above.
```suggestion
        # Separately add user to each additional group as Busybox's
        # 'adduser' does not support specifying additional groups.
        for addn_group in unsupported_busybox_values[
            "groups"
        ]:# pylint: disable=E1133
            LOG.debug("Adding user to group %s", addn_group)
            try:
                subp.subp(["addgroup", name, addn_group])
            except subp.ProcessExecutionError as e:
                    util.logexc(
                        LOG,
                        "Failed to add user %s to group %s",
                        name,
                        addn_group,
                    )
                    raise e
```

Reply:
Done